spring-cloud-vault-config-databases/src/test/java/org/springframework/cloud/vault/config/databases/VaultConfigCouchbaseDatabaseStaticTests.java

```diff
@@ -0,0 +1,144 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.vault.config.databases;
+
+import java.net.InetSocketAddress;
+
+import com.couchbase.client.java.*;
+import com.couchbase.client.java.kv.*;
+import com.couchbase.client.java.json.*;
+import com.couchbase.client.java.query.*;
+import com.couchbase.client.core.error.*;
+
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.Duration;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.vault.util.CanConnect;
+import org.springframework.cloud.vault.util.VaultRule;
+import org.springframework.cloud.vault.util.Version;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.vault.core.VaultOperations;
+
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Integration tests using the database secret backend. In case this test should fail
+ * because of SSL make sure you run the test within the
+ * spring-cloud-vault-config/spring-cloud-vault-config directory as the keystore is
+ * referenced with {@code ../work/keystore.jks}.
+ *
+ * Needs existing Couchbase user named vault-static with at least ro_admin privs.
+ *
+ * @author Francis Hitchens
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = VaultConfigCouchbaseDatabaseStaticTests.TestApplication.class,
+		properties = { "spring.cloud.vault.couchbase.enabled=true",
+				"spring.cloud.vault.couchbase.role=staticreadonly",
+				"spring.cloud.vault.couchbase.staticRole=true",
+				"spring.data.couchbase.username=foo",
+				"spring.data.couchbase.password=bar",
+				"spring.main.allow-bean-definition-overriding=true" })
+public class VaultConfigCouchbaseDatabaseStaticTests {
+
+	private static final int COUCHBASE_PORT = 8091;
+
+	private static final String COUCHBASE_HOST = "localhost";
+
+	@Value("${spring.data.couchbase.username}")
+	String username;
+
+	@Value("${spring.data.couchbase.password}")
+	String password;
+
+	Cluster cluster;
+
+	/**
+	 * Initialize the couchbase secret backend.
+	 */
+	@BeforeClass
+	public static void beforeClass() {
+
+		VaultRule vaultRule = new VaultRule();
+		vaultRule.before();
+
+		assumeTrue(CanConnect.to(new InetSocketAddress(COUCHBASE_HOST, COUCHBASE_PORT)));
+		assumeTrue(vaultRule.prepare().getVersion()
```

Comment:
In which version did Vault add Couchbase support? These guards are in place as we run tests against multiple Vault versions.

Reply:
The Couchbase plugin is not part of Vault, it is a independent of it but relies on the plugin SDK and Vault API. The API is V1.0.5 which is the only one available at present. The SDK version is V0.1.14-0. I guess these components must stay very stable as there are a lot of third party plugins.

I have only tested against Vault v1.4.0 and v1.4.1.